assets/charts/control-plane/calico-host-protection/templates/host-protection.yaml

```diff
@@ -1,15 +1,3 @@
-# General
-# =======
----
-# Disable default failsafe inbound rules
-apiVersion: crd.projectcalico.org/v1
-kind: FelixConfiguration
-metadata:
-  name: default
```

Comment:
Hm, as we move this object from `calico-host-protection` chart to `calico`, I guess we need to require from the user to remove this object manually before rolling an upgrade. i wonder if it has some implications like weakened rules for upgrade time (as failsafe port won't be emptied) OR that networking breaks for some reason.

I think we should test that.

Reply:
Tested the upgrade process and it involves deleting `default FelixConfiguration` while upgrading.

Reply:
I suggest we add "Release notes" section in PR description then, like in https://github.com/kinvolk/lokomotive/pull/946.

Reply:
By looking at the config we set and default values in Calico, following changes will be done during the upgrade:
- `bpfLogLevel` will be briefly set from `Info` to `Off`.
- Following ports will be opened on all hosts:
  ```
  - protocol: tcp
    port: 22
  - protocol: udp
    port: 68
  - protocol: tcp
    port: 179
  - protocol: tcp
    port: 2379
  - protocol: tcp
    port: 2380
  - protocol: tcp
    port: 5473
  - protocol: tcp
    port: 6443
  - protocol: tcp
    port: 6666
  - protocol: tcp
    port: 6667
  ```

I _think_ if we include that in release notes, it should be fine, given that this is only one time thing during the upgrade. Thoughts @iaguis @johananl ?

Reply:
@invidian done

Reply:
Isn't it becoming a lot of hassle for a released deemed as a `maintainence` release ?

Reply:
@ipochi what do you mean? The PR is almost complete, why wouldn't we merge it?

Reply:
I am not opposed to merging it.

The step of removal of felix configuration from its current chart of `calico-host-protection`, can this be avoided and not be touched for the scope of this PR ? Would that mean less hassles on the upgrade side in this release ?

Reply:
>The step of removal of felix configuration from its current chart of calico-host-protection, can this be avoided and not be touched for the scope of this PR ? Would that mean less hassles on the upgrade side in this release ?

Yes, it could possibly be avoided, but it will be messy. I think we should clean it up like this PR does. The upgrade steps are not so bad.

Reply:
Ok, I have no objections if others are okay with it.

Reply:
A couple of things.

* @ipochi not sure what you mean about a `maintenance` release. But if we include this in the next release it needs to be a minor release. Otherwise we'd be releasing a patch release that includes new features which is something we don't want.
* I find the syntax used when mentioning what ports are going to be opened during the upgrade weird, I'd use something like:

    * 22/tcp
    * 68/udp
    * ...

Otherwise I think it's ok to have these instructions for upgrades and have a small window of open ports if we warn about it. If we want to provide a more secure option, we could provide a temporary GlobalNetworkPolicy covering those ports and ask the user to apply it before the upgrade and remove it after.

Reply:
@iaguis updated description

Reply:
>Otherwise I think it's ok to have these instructions for upgrades and have a small window of open ports if we warn about it. If we want to provide a more secure option, we could provide a temporary GlobalNetworkPolicy covering those ports and ask the user to apply it before the upgrade and remove it after.

Just a note: temporary GNP won't work here, as safePorts bypasses it anyway as far as I understand.